src/pages/Product/Product.js

```diff
@@ -1,40 +1,78 @@
-import React from "react";
+import { useEffect } from "react";
+import axios from "axios";
+import "./product.css";
 import { Category } from "../../components/Filter/Category";
 import { Products } from "../../components/Products/Products";
 import { Navbar } from "../../components/Navbar/Navbar";
-import "./product.css";
-import { ProductData } from "../../components/Products/ProductData";
 import { Footer } from "../../components/Footer/Footer";
+import { useProduct } from "../../Context/Product-Context";
+import {
+  getSortedList,
+  getCategorizedData,
+  getPriceList,
+  getRating,
+} from "../../Reducer/Product-Reducer";
 const Product = () => {
+  const { state, dispatch } = useProduct();
+  const { initialValue } = state;
+  const getProductData = async () => {
+    const response = await axios.get("/api/products");
+    dispatch({ type: "FETCH_PRODUCT_DATA", payload: response.data.products });
+  };
```

Comment:
You could create this as a service. A service is all the network calls that you make to a server.
These could be added in `services` folder and the files having those should be named as `name.service.js`. e.g. `getProducts.service.js`